webpack.config.js

```diff
@@ -51,6 +55,20 @@ function getAliasesForExtensions() {
 	return aliasesMap;
 }
 
+/*
+ * Find package.json for file at `filepath`.
+ * Return `true` if it has a property whose key is `PROPKEY_ESNEXT` or 'module'.
+ */
+function hasPkgEsnext( filepath ) {
+	const pkgRoot = findRoot( filepath );
+	const packageJsonPath = path.resolve( pkgRoot, 'package.json' );
+	const packageJsonText = fs.readFileSync( packageJsonPath,
+		{ encoding: 'utf-8' } );
+	const packageJson = JSON.parse( packageJsonText );
+	return {}.hasOwnProperty.call( packageJson, PROPKEY_ESNEXT ) ||
+		{}.hasOwnProperty.call( packageJson, 'module' );
```

Comment:
why the awkward construction here vs `packageJson.hasOwnProperty( PROPKEY_ESNEXT )`?

Reply:
ha, did I find an un-cited reference for http://2ality.com/2017/06/pkg-esnext.html?

Reply:
Yes!

Reply:
should probably cite that

also, I'm still not sure _why_ we're doing this awkward thing - can you clarify?

Reply:
TBH I was blindly copy-pasting. But yeah, doesn't seem to make too much sense.

Reply:
+1 to simplifying. The only idea I could come up with is that it's guarding against packageJson not being an object.

Reply:
Could be. Anyway, can probably take the risk and simplify. :slightly_smiling_face: